### DISCUSSION TITLE
Error when get data type invoice

### DISCUSSION THREAD

**Discussion by Ferdinandgushcloud** [netsuitephp/netsuite-php#127] [Q&A]
Hi All, i tried a simple code to get invoice data with this code 

`$request = new GetRequest(); `
`$request->baseRef = new RecordRef();` 
`$request->baseRef->internalId = 7492; `
`$request->baseRef->type = "invoice"; `
`$getResponse = $service->get($request);`

error display `INVALID_TRANS_TYP. Transaction type specified is incorrect. `

when i check on the netsuite record type list from this link [Record Type List](https://www.netsuite.com/help/helpcenter/en_US/srbrowser/Browser2015_1/schema/enum/recordtype.html) , have type for invoice. 

can help me for this?

**Comment by jrebs** [comment]
Hi Ferdinand. It sounds to me like perhaps the record that you are trying to get is not an invoice. Do you know the difference between an *invoice ID* and an *internal ID* for that same record? My hunch here is that 7492 is an invoice number and not the internalId of the record it refers to. If you look up the record in question in the NetSuite web interface, look in the address bar and you should see the actual internalId of the record shown there. Try using this number in your code. If I'm mistaken and you're definitely using the internal ID already, then I guess that hunch was wrong and I have no other suggestion without some more information.
